### PR TITLE
Use nodesource repositories version 12 on slaves

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: puppet/modules/slave/manifests/init.pp [hunk 1]
```diff
@@ -150,13 +150,13 @@
...
   if $::osfamily == 'RedHat' {
-    package { 'npm':
-      ensure  => present,
-      require => Package['bzip2'],
-    } -> Package <| provider == 'npm' |>
+    class { 'nodejs':
+      repo_url_suffix       => '12.x',
+      nodejs_package_ensure => latest,
+      npm_package_ensure    => latest,
+    }
```
Comment (mmoll) [inline puppet/modules/slave/manifests/init.pp]: `-> Package <| provider == 'npm' |>` is IMHO still needed, or does the nodejs class handle this on its own?

Comment (ekohl) [reply puppet/modules/slave/manifests/init.pp]: Updated

File: puppet/modules/slave/manifests/init.pp [hunk 2]
```diff
@@ -150,12 +150,12 @@
...
-  # packages only really available on EL6+
+  # nodejs/npm for JavaScript tests
   if $::osfamily == 'RedHat' {
-    package { 'npm':
-      ensure  => present,
-      require => Package['bzip2'],
+    class { 'nodejs':
+      repo_url_suffix       => '12.x',
+      nodejs_package_ensure => latest,
+      npm_package_ensure    => latest,
```
Comment (mmoll) [inline puppet/modules/slave/manifests/init.pp]: needs to be `absent`, as the `nodejs` package is already containing it.

Comment (ekohl) [reply puppet/modules/slave/manifests/init.pp]: Updated. Also explicitly set `manage_repo` to true.

Comment (mmoll) [reply puppet/modules/slave/manifests/init.pp]: `Class[Nodejs]: has no parameter named 'manage_repo'` :upside_down_face:

**Comment by mmoll** [comment]
merged, bedankt @ekohl!

**Pull request merged**